app/models/book.rb

```diff
@@ -39,7 +48,7 @@ def create_authors authors_params
     cascade_authors
     self.authors.clear
     authors_params[:names].split(',').each do |value|
-      author = Author.find_or_initialize_by({name: value})
+      author = Author.find_or_initialize_by({name: value.split.join(" ")})
```

Comment:
split?

Reply:
to avoid extra spaces and ensure the name is unique

Reply:
if it is only the white spaces use `strip` method, if it is more complex than that give regex a try.

Reply:
is this look better `str.strip.gsub(/\s+/, " ")`?